Sources/AppcuesKit/Presentation/CustomComponents/AppcuesExperienceActions.swift

```diff
@@ -65,7 +65,9 @@ public class AppcuesExperienceActions {
     }
 
     private func enqueue(_ action: AppcuesExperienceAction) {
-        guard let actionRegistry = appcues?.container.resolve(ActionRegistry.self) else { return }
-        actionRegistry.enqueue(actionInstances: [action]) {}
+        Task {
+            guard let actionRegistry = appcues?.container.resolve(ActionRegistry.self) else { return }
+            try await actionRegistry.enqueue(actionInstances: [action])
+        }
```

Comment:
So this is interesting. You absolutely could get actions from multiple user interactions executing out of order (of course the actions for a single user interaction will always execute in order).

`actionRegistry.enqueue` is a misnomer—it's not adding the items to the queue, it's just executing them. We do the same for push actions and pre-step navigation actions where they just execute immediately. At the very least this function should be renamed to not be `enqueue`. But I think there's a good opportunity to refactor further and use an actor and `AsyncStream` instead of janky `isProcessing` flag.